plugins/prometheus_plugin/prometheus_plugin.cpp

```diff
@@ -214,24 +220,35 @@ namespace eosio {
    prometheus_plugin::~prometheus_plugin() = default;
 
    void prometheus_plugin::set_program_options(options_description&, options_description& cfg) {
+      cfg.add_options()
+         ("prometheus-exporter-address", bpo::value<string>()->default_value("127.0.0.1:9101"),
+            "The local IP and port to listen for incoming prometheus metrics http request.");
    }
 
    void prometheus_plugin::plugin_initialize(const variables_map& options) {
       my->initialize_metrics();
 
-      auto& _http_plugin = app().get_plugin<http_plugin>();
-      fc::microseconds max_response_time = _http_plugin.get_max_response_time();
+      string lipstr = options.at("prometheus-exporter-address").as<string>();
+      EOS_ASSERT(lipstr.size() > 0, chain::plugin_config_exception, "prometheus-exporter-address must have a value");
+
+      string host = lipstr.substr(0, lipstr.find(':'));
+      string port = lipstr.substr(host.size() + 1, lipstr.size());
+
+      boost::system::error_code ec;
+      using tcp = boost::asio::ip::tcp;
+      tcp::resolver resolver(app().get_io_service());
 
-      prometheus_api handle(*my, max_response_time);
-      app().get_plugin<http_plugin>().add_async_api({
-        CALL_ASYNC_WITH_400(prometheus, handle, eosio, metrics, std::string, 200, http_params_types::no_params)}, http_content_type::plaintext);
+      my->_endpoint = *resolver.resolve(tcp::v4(), host, port, ec);
```

Comment:
Is there some reason we can't support IPv6?

Reply:
Our existing HTTP plugin use ipv4 only and I couldn't find any example from goole which listens an interface returned from tcp::resolver with both ipv4 and ipv6 enabled. I was just not confident that I won't introduce unforeseeable bugs if I dit that. If we want to support ipv6, I believe we should wait until 5.0 and do it with the existing http plugin.

Reply:
Yeah, I think holding off on IPv6 is the right choice until there is more reflection on how to handle the various options discussed in #622